### PR TITLE
acronym: Update to most recent spec

### PR DESCRIPTION
**Pull request by wolf99**
- Add missing test cases
  - Update implementation to handle leading underscores for one of these new cases
- Move C-specific extra test cases (null and empty string) to top
- Remove outdated test cases
- Refactor helper function to indicate it is a check rather than a test (in line with practice in other exercises)

### CODE REVIEW

**Comment by wolf99** [comment]
Travis seems to be intermittently failing with:

```
The command "bin/fetch-configlet" exited with 2.
0.00s$ bin/configlet lint .
/home/travis/.travis/functions: line 109: bin/configlet: No such file or directory
The command "bin/configlet lint ." exited with 127.
```

Any clues why?

**Comment by iHiD** [comment]
(cc @ErikSchierboom - I'm seeing this travis issue in lots of places)

File: exercises/acronym/src/example.c [hunk 1]
```diff
@@ -6,7 +6,8 @@ static int is_first_letter_of_word(int character, int preceding_character)
 {
    return isalpha(character)
        && ((preceding_character == ' ')
-           || (preceding_character == '-')) ? 1 : 0;
+           || (preceding_character == '-')
+           || (preceding_character == '_')) ? 1 : 0;
```
Comment (ryanplusplus) [inline exercises/acronym/src/example.c:10]: Is the `? 1 : 0` actually necessary?

Comment (ryanplusplus) [reply exercises/acronym/src/example.c:10]: I'm fairly certain the answer is no. We should probably tweak this to return a `bool` for clarity.
